sdk/core/azure_core/src/response.rs

```diff
@@ -46,189 +93,88 @@ impl RawResponse {
+    // However, we ALSO want to allow users to specify a different target type if they want to (for example, to limit what they deserialize or work around a service-specific issue).
+    // So this method is a compromise that allows most users to avoid specifying the target type while still allowing users to specify a different target type if they want to.
+    // If Rust ever stabilizes type parameter defaults for functions (https://github.com/rust-lang/rust/issues/27336), we could consider changing this.
+    pub fn map_body<U>(self) -> Response<U> {
+        Response {
+            status: self.status,
+            headers: self.headers,
+            body: self.body,
             phantom: PhantomData,
         }
     }
 }
 
-pub struct Response<T> {
-    response: RawResponse,
-    phantom: PhantomData<T>,
-}
-
-impl<T> Response<T> {
-    /// Create an HTTP response.
-    pub fn new(status: StatusCode, headers: Headers, stream: PinnedStream) -> Self {
-        Self {
-            response: RawResponse::new(status, headers, stream),
-            phantom: PhantomData,
-        }
-    }
-
-    /// Get the status code from the response.
-    pub fn status(&self) -> StatusCode {
-        self.response.status
-    }
-
-    /// Get the headers from the response.
-    pub fn headers(&self) -> &Headers {
-        &self.response.headers
-    }
-
-    /// Deconstruct the HTTP response into its components.
-    pub fn deconstruct(self) -> (StatusCode, Headers, ResponseBody) {
-        (
-            self.response.status,
-            self.response.headers,
-            self.response.body,
-        )
-    }
-
-    /// Consume the HTTP response and return the HTTP body bytes.
-    pub fn into_body(self) -> ResponseBody {
-        self.response.body
-    }
-
-    /// Get the response body as the specified type from JSON.
-    pub async fn json(self) -> crate::Result<T>
-    where
-        T: DeserializeOwned,
-    {
-        self.into_body().json().await
-    }
-
-    /// Get the response body as the specified type from XML.
-    #[cfg(feature = "xml")]
-    pub async fn xml(self) -> crate::Result<T>
-    where
-        T: DeserializeOwned,
-    {
-        self.into_body().xml().await
+impl<T: FromResponseBody> Response<T> {
+    /// Fetches the entire body and tries to convert it into type `T`.
+    pub async fn read_body(self) -> crate::Result<T> {
+        T::from_response_body(self.body).await
     }
 }
 
 impl<T> fmt::Debug for Response<T> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.debug_struct("Response")
-            .field("status", &self.response.status)
-            .field("headers", &self.response.headers)
+            .field("status", &self.status)
+            .field("headers", &self.headers)
             .field("body", &"(body)")
             .finish()
     }
 }
 
-/// An HTTP response with the body collected as bytes.
-#[derive(Debug, Clone)]
-pub struct CollectedResponse<T> {
```

Comment:
The `main` branch was using it, though off hand I can't remember why. I think it was primarily for testing and/or pageables.

Reply:
Do you feel a need to have it back? I'd like to avoid having a separate type to represent this, if possible. If/when we need to represent a response using collected bytes (instead of a stream) we can revisit this.